src/main/java/org/opensearch/security/securityconf/DynamicConfigModelV7.java

```diff
@@ -383,7 +386,7 @@ private void buildAAA() {
          * order: -1 - prioritize the OBO authentication when it gets enabled
          */
         Settings oboSettings = getDynamicOnBehalfOfSettings();
-        if (oboSettings.get("signing_key") != null && oboSettings.get("encryption_key") != null) {
+        if (!isSigningKeyNull(oboSettings) && !isEncryptionKeyNull(oboSettings)) {
```

Comment:
```suggestion
        if (!isNull(oboSettings, "signing_key") && !isNull(oboSettings, "encryption_key")) {
```

Reply:
You can use `Strings.isNullOrEmpty`. No need to re-invent the wheel, this method is used pretty extensively.

Why not encapsulate the logic into a higher-level method that captures the intent of the code. i.e. `public static OBOSettings.isValid(Settings oboSettings)` or something like that and capture any specific logic within that method so that the same logic is not repeated in multiple places and if there's an update to the logic only a single place needs updating?